cli/cmd/endpoints.go

```diff
@@ -38,45 +50,44 @@ func (o *endpointsOptions) validate() error {
 
 func newEndpointsOptions() *endpointsOptions {
 	return &endpointsOptions{
-		namespace:    "",
 		outputFormat: tableOutput,
 	}
 }
 
 func newCmdEndpoints() *cobra.Command {
 	options := newEndpointsOptions()
 
-	example := `  # get all endpoints
-  linkerd endpoints
+	example := `  # get all endpoints for the authorities emoji-svc.emojivoto.svc.cluster.local:8080 and web-svc.emojivoto.svc.cluster.local:80
+  linkerd endpoints emoji-svc.emojivoto.svc.cluster.local:8080 web-svc.emojivoto.svc.cluster.local:80
 
-  # get endpoints in the emojivoto namespace
-  linkerd endpoints -n emojivoto
+  # get that same information in json format
+  linkerd endpoints -o json emoji-svc.emojivoto.svc.cluster.local:8080 web-svc.emojivoto.svc.cluster.local:80
 
-  # get all endpoints in json
-  linkerd endpoints -o json`
+  # get the endpoints for authorities in Linkerd's control-plane itself
+  linkerd endpoints linkerd-controller-api.linkerd.svc.cluster.local:8085
+  linkerd endpoints linkerd-web.linkerd.svc.cluster.local:8084`
 
 	cmd := &cobra.Command{
-		Use:     "endpoints [flags]",
+		Use:     "endpoints [flags] authorities",
 		Aliases: []string{"ep"},
 		Short:   "Introspect Linkerd's service discovery state",
 		Long: `Introspect Linkerd's service discovery state.
 
 This command provides debug information about the internal state of the
-control-plane's destination container. Note that this cache of service discovery
-information is populated on-demand via linkerd-proxy requests. This command
-will return "No endpoints found." until a linkerd-proxy begins routing
-requests.`,
+control-plane's destination container. It queries the same Destination service
```

Comment:
The output and functionality of the command is the same, the only difference being that now we're giving the option to filter by authority(ies)

Reply:
i think this new comment is accurate. there's a subtle change where the previous behavior of `linkerd endpoints` would just dump the state of the cache in service discovery. the new behavior retrieves endpoint data the same as a `linkerd-proxy` would, so if it's in the cache, it will get from there, but if not in the cache, it will initiate a watch and populate the cache. does this sound right?